include/errno.h

```diff
@@ -17,6 +17,7 @@
 #define _LIBPHOENIX_ERRNO_H_
 
 #include <phoenix/errno.h>
+#include <arch.h>
```

Comment:
`errno.h` should not include `arch.h`, please note that `arch.h` is needed only for errno implementation, so it should be included in `errno.c`.

IMHO `SUPPORT_TLS` name is also too generic, might be set by accident by other packages (eg. mbedTLS ;)). Not sure we have the convention for internal defines in `libphoenix`, maybe prepend it with `__LIBPHOENIX_` (eg. `__LIBPHOENIX_ARCH_TLS_ENABLED` (or SUPPORTED) ?)

Reply:
Fixed